homeassistant/components/logbook.py

```diff
@@ -317,43 +379,70 @@ def humanify(hass, events):
                 }
 
 
+def _get_related_entity_ids(session, events_filter):
+    from homeassistant.components.recorder.models import States
+    from homeassistant.components.recorder.util import \
+        RETRIES, QUERY_RETRY_WAIT
+    from sqlalchemy.exc import SQLAlchemyError
+    import time
+
+    timer_start = time.perf_counter()
+
+    query = session.query(States).with_entities(States.domain,
+                                                States.entity_id).distinct()
+
+    for tryno in range(0, RETRIES):
```

Comment:
we have a helper for that

Reply:
do you mean `execute` from homeassistant/components/recorder/util.py ?
it does not fit in here as i want to work with raw records

Reply:
We should consider generalizing our helper then.